### PR TITLE
Add AVRO map type into jsonb

### LINKED ISSUE #2213: Document: feat(source): parse avro map into jsonb
This issue tracks the documentation update needed for the merged PR #16948.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/16948
Source PR Merged At: 2024-05-28T08:39:04Z

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Info

- **Description**

  - [ What's changed? Which parts of the docs are affected? ]

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/16948

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/2213#event-12967861676

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

File: docs/ingest/supported-sources-and-formats.md [hunk 1]
```diff
@@ -72,6 +72,16 @@ ENCODE AVRO (
...
+In addition, you can use the option `map.handling.mode` to ingest AVRO map type into JSONB. For example:
+
+```sql
+FORMAT [ DEBEZIUM | UPSERT | PLAIN [, ...] ] ENCODE AVRO (
+	map.handling.mode = 'jsonb'
+)
+```
+
+Note that the value types can only be: `null`, `boolean`, `int`, `string`, or `map`/`record`/`array` with these types.
+
```
Comment (ShanlanLi) [inline docs/ingest/supported-sources-and-formats.md:84]: I update this new feature under the heading "AVRO", not very sure if this place is suitable. Please tell me if there is a better choice.
image

Comment (xxchan) [reply docs/ingest/supported-sources-and-formats.md:84]: I think maybe we can document "FORMAT" (JSON/AVRO/..) and "ENCODING" (PLAIN/UPSERT/DEBEZIUM) separately. cc @xiangjinwu

**Review by xxchan — APPROVED** [review]
not too bad. We can refactor later

File: docs/ingest/supported-sources-and-formats.md (hunk 1 above)
Comment (xiangjinwu) [reply docs/ingest/supported-sources-and-formats.md:84]: It was legacy from the old `row format` syntax I guess ...

File: docs/ingest/supported-sources-and-formats.md [hunk 2]
```diff
@@ -72,6 +72,16 @@ ENCODE AVRO (
 )
 ```
 
+In addition, you can use the option `map.handling.mode` to ingest AVRO map type into JSONB. For example:
+
+```sql
+FORMAT [ DEBEZIUM | UPSERT | PLAIN [, ...] ] ENCODE AVRO (
```
Comment (xiangjinwu) [inline docs/ingest/supported-sources-and-formats.md]: What is this `[, ...]` ?

Comment (ShanlanLi) [reply docs/ingest/supported-sources-and-formats.md]: Do I need to delete ` [, ...]`? Or just copy this one:
image

File: docs/ingest/supported-sources-and-formats.md (hunk 1 above)
Comment (ShanlanLi) [reply docs/ingest/supported-sources-and-formats.md:84]: @xiangjinwu Then do we need to refactor this page?

File: docs/ingest/supported-sources-and-formats.md (hunk 2 above)
Comment (xiangjinwu) [reply docs/ingest/supported-sources-and-formats.md]: `...` in the Release note means one of `[ DEBEZIUM | UPSERT | PLAIN ]` and either form looks good to me. But `PLAIN [, ...]` is definitely wrong and means multiple comma-separated items can be specified.

File: docs/ingest/supported-sources-and-formats.md (hunk 1 above)
Comment (xiangjinwu) [reply docs/ingest/supported-sources-and-formats.md:84]: Yes but in a separate PR. Thank you!

**Review by WanYixian — APPROVED** [review]

File: docs/ingest/supported-sources-and-formats.md (hunk 1 above)
Comment (xiangjinwu) [reply docs/ingest/supported-sources-and-formats.md:84]: WIP: #2294

**Pull request merged**